### PR TITLE
Enhancement Input Device API 

### PR DESCRIPTION
**Pull request by DavidMansolino**
Linked to omichel/webots#7304

https://www.cyberbotics.com/doc/reference/robot?version=enhancement.input-device-api#wb_robot_wait_for_user_input_event

https://www.cyberbotics.com/doc/reference/cpp-api?version=enhancement.input-device-api#cpp_robot
https://www.cyberbotics.com/doc/reference/java-api?version=enhancement.input-device-api#java_robot
https://www.cyberbotics.com/doc/reference/python-api?version=enhancement.input-device-api#python_robot
https://www.cyberbotics.com/doc/reference/matlab-api?version=enhancement.input-device-api#matlab_robot
https://www.cyberbotics.com/doc/reference/ros-api?version=enhancement.input-device-api#robot

### CODE REVIEW

**Comment by DavidMansolino** [comment]
👍

**Comment by DavidMansolino** [comment]
`wb_robot_wait_for_user_input` is fine for me, then I assume we will also rename the events like this: `MOUSE_MOVE_EVENT` => `MOUSE_MOVE_INPUT` ?

**Comment by DavidMansolino** [comment]
👍   I would like to have an intermediate review before scaling this to the other API: https://www.cyberbotics.com/doc/reference/robot?version=enhancement.input-device-api#wb_robot_wait_for_user_input_event

**Comment by tn12787** [comment]
👀

File: reference/robot.md [hunk 1]
```diff
@@ -343,6 +343,52 @@ for(i=0; i<n_devices; i++) {
...
+
+```c
+#include <webots/robot.h>
+
+int wb_robot_wait_for_user_input_event(int event_type, int timeout);
+```
+
+**Description**
+
+This function can be used to get [Joystick](joystick.md), [Keyboard](keyboard.md) and [Mouse](mouse.md) input without calling the `wb_robot_step` function, this is usefull to prevent the simulation from running until a specific user input event happens.
```
Comment (tn12787) [inline reference/robot.md]: usefull -> useful

**Comment by omichel** [comment]
The function and constant naming looks good to me.

**Comment by DavidMansolino** [comment]
👍

**Comment by DavidMansolino** [comment]
👎   I will reflect the latest changes from omichel/webots#7304

**Comment by DavidMansolino** [comment]
👍

**Pull request merged**